scripts/mbedtls_dev/crypto_test_maker.py

```diff
@@ -0,0 +1,130 @@
+"""Generate unit test cases for Mbed TLS crypto functions.
+
+Functions in this module are aligned with test functions in
+``tests/suites/*.function`` at the time of writing, and their output is
+suitable for corresponding ``.data`` files. However this module is not
+meant to be complete and is not kept up to date.
+"""
+
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+from typing import List, Optional, Union
+
+from mbedtls_dev import crypto_knowledge
+from mbedtls_dev import crypto_reference
+from mbedtls_dev import test_case
+
+
+class Common(test_case.TestCase):
+    """Helper code for test function simulators."""
+
+    def __init__(self, function: str, *args: str) -> None:
+        """Dynamic dispatch to the method whose name is function.
+
+        Extra args are passed to the function. If args is empty, function can
+        be a colon-separated sequence where the first element is the function
+        name, as in a data line in a .data file.
+
+        The method must return a list of arguments which will be appended
+        to the input arguments.
+        """
+        super().__init__()
+        function = function.strip()
+        if len(args) == 0 and ':' in function:
+            parts = function.split(':')
+            function = parts[0]
+            args = tuple(arg for arg in parts[1:] if arg)
+        outputs = getattr(self, function)(*args)
```

Comment:
While testing I noticed that script crushes when arguments are invalid (unsupported function name is provided, argument is missing, invalid input - no crush, but also no error indication). Maybe this is not very important and user should ensure that given input is valid, but maybe we could verify only if number of arguments is correct and given function is supported before calling this function?
If error is caught then output could hold the error info.
```
sss:PSA_ALG_HKDF(PSA_ALG_SHA_256):"0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b":"000102030405070708090a0b0c":"f0f1f2f3f4f5f6f7f8f9":PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1):256:
! Error at input line 1:
Traceback (most recent call last):
  File "./tests/scripts/complete_test_case.py", line 58, in <module>
    main(sys.argv[1:])
  File "./tests/scripts/complete_test_case.py", line 49, in main
    tc = suite_class(line)
  File "/home/przemek/src/mbedtls/tests/scripts/../../scripts/mbedtls_dev/crypto_test_maker.py", line 51, in __init__
    outputs = getattr(self, function)(*args)
AttributeError: 'PsaCrypto' object has no attribute 'sss'

przemek@przemek-VirtualBox:~/src/mbedtls$ ./tests/scripts/complete_test_case.py psa_crypto
derive_key_type:PSA_ALG_HKDF(PSA_ALG_SHA_256)::"000102030405070708090a0b0c":"f0f1f2f3f4f5f6f7f8f9":PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1):256:
! Error at input line 1:
Traceback (most recent call last):
  File "./tests/scripts/complete_test_case.py", line 58, in <module>
    main(sys.argv[1:])
  File "./tests/scripts/complete_test_case.py", line 49, in main
    tc = suite_class(line)
  File "/home/przemek/src/mbedtls/tests/scripts/../../scripts/mbedtls_dev/crypto_test_maker.py", line 51, in __init__
    outputs = getattr(self, function)(*args)
TypeError: derive_key_type() missing 1 required positional argument: 'bits'

przemek@przemek-VirtualBox:~/src/mbedtls$ ./tests/scripts/complete_test_case.py psa_crypto
test
test

```